src/dotnet/App.Maui/Platforms/Android/MainActivity.cs

```diff
@@ -129,9 +148,29 @@ protected override void OnNewIntent(Intent? intent)
         TryProcessNotificationTap(intent);
     }
 
+    public void RequestPermissions(string permission)
+        => _requestPermissionLauncher.Launch(permission);
+
+    public override void OnRequestPermissionsResult(int requestCode, string[] permissions, Permission[] grantResults)
+    {
+        base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
+        if (requestCode == NotificationPermissionID) {
+            var (_, notificationGrant) = permissions
+                .Zip(grantResults)
+                .FirstOrDefault(tuple => tuple.First == Manifest.Permission.PostNotifications);
+            var notificationState = notificationGrant switch {
+                Permission.Denied => PermissionState.Denied,
+                Permission.Granted => PermissionState.Granted,
+                _ => throw new ArgumentOutOfRangeException(),
+            };
+            var notificationUi = AppServices.GetRequiredService<NotificationUI>();
```

Comment:
don't we use UI suffix? like notificationUI

Reply:
Yep, `Ui` is a wrong one, it must be `UI`: both letters in two-letter acronyms on .NET should be capitalized: https://stackoverflow.com/questions/2109171/c-sharp-naming-conventions-for-acronyms

The only exception we have is `Db`, where the rule was broken by .NET team itself.